core/src/main/kotlin/io/specmatic/core/pattern/JSONObjectPattern.kt

```diff
@@ -505,6 +505,16 @@ data class JSONObjectPattern(
         }
     }
     
+    /**
+     * Checks if a path needs to be wrapped in braces.
+     * Returns true for simple identifiers (like typeAlias names or scalar type names).
+     * Simple identifiers are strings that start with a letter and contain only letters and numbers.
+     * This includes scalar type names like "string", "number", and "boolean".
+     */
+    private fun needsBraces(path: String): Boolean {
+        return path.isNotEmpty() && path.first().isLetter() && path.all { it.isLetterOrDigit() }
```

Comment:
Checking for first letter and checking for all are overlapping. Update the code to check for all. 

And when done, check the rest of this file to see if other code needs to use this function.